crates/consensus/src/transaction/rlp.rs

```diff
@@ -109,14 +109,14 @@ pub trait RlpEcdsaEncodableTx: Sized {
 
     /// Calculate the transaction hash for the given signature.
     fn tx_hash(&self, signature: &Signature) -> TxHash {
-        self.tx_hash_with_type(signature, Self::DEFAULT_TX_TYPE)
+        self.tx_hash_with_type(signature, self.ty())
     }
 }
 
 /// Helper trait for managing RLP decoding of transactions inside 2718 envelopes.
 #[doc(hidden)]
 #[doc(alias = "RlpDecodableTx", alias = "RlpTxDecoding")]
-pub trait RlpEcdsaDecodableTx: RlpEcdsaEncodableTx {
+pub trait RlpEcdsaDecodableTx: RlpEcdsaEncodableTx + Typed2718 {
```

Comment:
here we now need to introduce the associated constant for now, because we can't remove that in one go, but we want to remove it from the encodable trait right away